src/couch_file.erl

```diff
@@ -247,7 +247,7 @@ delete_file(FilePath) ->
 deleted_filename(_RootDir, Original, rename) ->
     deleted_filename(Original, "deleted");
 deleted_filename(RootDir, Original, _Strategy) ->
-    DelFileName = deleted_filename_base(RootDir, Original),
+    DelFileName = couch_util:url_encode(Original -- RootDir),
```

Comment:
I have a concern with using `--` here. I know that it is a common thing in erlang community to do something like this. But it so error prone. There are at least two problems:
1. number of slashes makes the difference
2. it brakes badly if two paths do not have common prefix
```
"/foo/bar/baz" -- "/src/bar".
"foo/baz"
```
Feel free to ignore this. Since this is my complain to the state of erlang's stdlib which is the mess.